### PR TITLE
feat(profile): add incoming follow requests under current user profile

### PR DESCRIPTION
**Pull request by kevincador**
## Summary

Related to #2459.

Adds incoming follow request support to profile social lists for private users. When the signed-in user has at least one pending incoming follow request, the profile social toggler now shows a `Requests` state. Selecting it renders the requesting users in the same profile row/card UI used by followers and following, so the user can open the requester profile.

The requester profile overflow menu now exposes `Approve Request` and `Reject Request` actions, with confirmation snackbars after successful approval or rejection.

## Details

- Adds a `currentUserFollowRequestsQuery` for `GET /users/requests` with full user mapping and follow-state invalidation.
- Adds approve and deny request helpers for `POST /users/requests/:id` and `DELETE /users/requests/:id`.
- Shows the `Requests` social toggle only for the signed-in user’s own profile and only when incoming requests exist.
- Falls back to `following` if the current social toggle becomes hidden after requests are cleared.
- Busts follow caches after follow-state changes so following, followers, pending follows, and incoming requests refetch consistently.
- Adds approve/reject actions to profile overflow menus only when that profile has an incoming request.
- Shows success snackbars only after the API confirms the approve/reject action.
- Allows the shared `Snackbar` component to be used as a notification-only toast without an action button.
- Adds MSW mocks and focused tests for incoming requests and follow invalidation keys.

## Edge Cases

- Non-authenticated users and other users’ profiles do not fetch or show incoming follow requests in the social toggler.
- The `Requests` toggle is hidden when the incoming request list is empty.
- If an approve/reject API response does not match the expected success status, the snackbar is not shown and caches are not invalidated.
- Denying a request expects `204`; approving expects `200`.
- Existing follow/unfollow/request-cancel behavior continues to share the same follow invalidation path.

## Flow

Follow a private profile (B) with account A.
Screenshot 2026-06-10 at 13 32 04

Shows as Pending under A's account.
Screenshot 2026-06-10 at 13 31 44

🆕 A's request is displayed under B's account.
Screenshot 2026-06-10 at 13 32 33

🆕 B can approve or reject the request.
Screenshot 2026-06-10 at 13 32 45

Snackbar without action is displayed for action confirmation.
Screenshot 2026-06-10 at 13 34 24

A's profile appears in B's Followers list 🎉
Screenshot 2026-06-10 at 13 34 45

### CODE REVIEW

**Comment by vladjerca** [comment]
@kevincador would be cool to add some quick ✅ action on the avatar here 

image 

would be especially handy for mouse users 👀

File: projects/client/src/lib/sections/profile/components/ProfileListPaginated.svelte [hunk 1]
```diff
@@ -10,15 +11,17 @@
...
   const { profiles, isLoading } = $derived(useFollowing(slug, type));
 
   const placeholder = $derived(
     type === "following"
       ? m.list_placeholder_following()
-      : m.list_placeholder_followers(),
+      : type === "followers"
+        ? m.list_placeholder_followers()
+        : m.list_placeholder_follow_requests(),
   );
```
[the remark this answers is not in the document]
Comment (vladjerca) [reply projects/client/src/lib/sections/profile/components/ProfileListPaginated.svelte]: @gemini-code-assist Adopted - extracted `getProfileSocialPlaceholder(type)` helper to `lib/sections/profile/_internal/`. Same helper now used by ProfilesList.svelte too.

File: projects/client/src/lib/sections/profile/components/ProfilesList.svelte [hunk 2]
```diff
@@ -16,13 +16,15 @@
...
   const { profiles, isLoading } = $derived(useFollowing(slug, $current.value));
 
   const placeholder = $derived(
     $current.value === "following"
       ? m.list_placeholder_following()
-      : m.list_placeholder_followers(),
+      : $current.value === "followers"
+        ? m.list_placeholder_followers()
+        : m.list_placeholder_follow_requests(),
   );
```
[the remark this answers is not in the document]
Comment (vladjerca) [reply projects/client/src/lib/sections/profile/components/ProfilesList.svelte]: @gemini-code-assist Adopted - shares `getProfileSocialPlaceholder(type)` helper with ProfileListPaginated.svelte.

File: projects/client/src/lib/components/snackbar/Snackbar.svelte [hunk 3]
```diff
@@ -54,15 +59,17 @@
...
+    <Button
+      size="small"
+      variant="primary"
+      color="purple"
+      label={action.label}
+      onclick={action.onClick}
+    >
+      {action.text}
+    </Button>
+  {/if}
```
[the remark this answers is not in the document]
Comment (vladjerca) [reply projects/client/src/lib/components/snackbar/Snackbar.svelte:83]: @gemini-code-assist Leaving as-is - branch was rebased onto main since this review. Mobile path now renders `actionButton` without an outer `{#if action}` guard, so the inner check stays load-bearing for that branch. The popover path keeps its outer guard (it wraps additional markup), and reuses the same snippet.

**Review by seferturan — APPROVED** [review]
